app/experimenter/nimbus-ui/src/lib/config.ts

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { deepMerge } from './utilities';
+
+export interface Config {
+  sentry_dsn: string;
+  version: string;
+}
+
+export function getDefault() {
+  return {
+    sentry_dsn: '',
+    version: '',
+  }
+}
+
+export function readConfig(
+  mountingElement: HTMLElement
+) {
+  const content = mountingElement.dataset.config;
+  update(decode(content));
+}
+
+export function decode(content?: string) {
+  const isDev = process.env.NODE_ENV === 'development';
+
+  if (!content) {
+    if (isDev) {
+      console.warn('Nimbus is missing server config');
+    } else {
+      throw new Error('Configuration is empty');
+    }
+  }
+
+  const decoded = decodeURIComponent(content!);
+
+  try {
+    return JSON.parse(decoded);
+  } catch (error) {
+    if (isDev) {
+      console.warn('Nimbus server config is invalid');
+    } else {
+      throw new Error(
+        `Invalid configuration ${JSON.stringify(content)}: ${decoded}`
+      );
+    }
+  }
+}
+
+export function reset() {
+  const initial = getDefault();
+
+  // This resets any existing default
+  // keys back to their original value
+  Object.assign(config, initial);
+
+  // This removes any foreign keys that
+  // may have found there way in
+  Object.keys(config).forEach((key) => {
+    if (!initial.hasOwnProperty(key)) {
+      delete (config as any)[key];
+    }
+  });
+}
+
+export function update(newData: { [key: string]: any }) {
+  deepMerge(config, newData);
+}
+
+const config: Config = getDefault();
```

Comment:
So this treats `config` as a mutable global singleton?  Is there not a way to just pass it in to the react init as a paramter or a prop or something?

Reply:
It could certainly be passed in to the React app, but then you need to continue passing it down to each component that uses or has children that use it. We could also store it in Context if we wanted to avoid the whole component tree. Open to suggestions. Tagging @LZoog as well.

Reply:
Right yeah, the ol prop hot potato pattern.  I've attached things to `window` to solve this in the past, like `window.nimbus_config` or some such, that way it's easy to poke at while debugging and it's obvious where it's being stored, but this works too.

Reply:
I'd like to avoid attaching global properties to `window` if possible but I also know sometimes that can be useful. We're doing something similar for `fxa-settings` so I think leaving it is fine🤔